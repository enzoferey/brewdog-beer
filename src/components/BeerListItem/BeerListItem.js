import React from "react";
import PropTypes from "prop-types";
import style from "./beer-list-item.scss";

import getAbvLabel from "utils/getAbvLabel";
import getMaxLengthText from "utils/getMaxLengthText";

const MAX_LENGTH_DESCRIPTION = 360;

const BeerListItem = ({ background, image, name, abv, description }) => (
  <div className={style.main} style={{ background }}>
    <div className={style.photo}>
      <img src={image} alt="Beer bottle image" />
    </div>
    <div className={style.details}>
      <h4 className={style.name}>{name}</h4>
      <span className={style.attributes}>{getAbvLabel(abv)}</span>
      <p className={style.description}>
        {getMaxLengthText(description, MAX_LENGTH_DESCRIPTION)}
      </p>
    </div>
  </div>
);

BeerListItem.propTypes = {
  background: PropTypes.string,
  image: PropTypes.string,
  name: PropTypes.string,
  abv: PropTypes.number,
  description: PropTypes.string,
};

export default BeerListItem;
