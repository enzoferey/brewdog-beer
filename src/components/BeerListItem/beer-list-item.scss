$height: 290px;

.main {
  width: 100%;
  height: $height;
  display: flex;
  flex-wrap: wrap;
  color: white;

  &:hover {
    .photo > img {
      transform: rotate(10deg) scale(1.1);
    }

    .photo {
      width: calc(#{$height} + 15px);
    }
  }
}

.photo {
  position: relative;
  width: $height;
  height: $height;
  padding: 43px 0;
  box-sizing: border-box;
  text-align: center;
  background-color: rgba(white, 0.1);
  transition: all 0.3s;

  > img {
    height: 100%;
    transition: all 0.3s;
  }
}

.details {
  flex: 1;
  height: 100%;
  padding: 32px 40px;
  box-sizing: border-box;
}

.name {
  font-size: 32px;
  margin: 0 0 5px;
}

.attributes {
  font-size: 24px;
  color: #eeeeee;
}

.description {
  font-size: 20px;
  margin: 20px 0 0;
}
