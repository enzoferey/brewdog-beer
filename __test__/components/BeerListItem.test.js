import React from "react";
import BeerListItem from "components/BeerListItem";

const props = {
  background: "pink",
  image: "src-path",
  name: "Beer name",
  abv: 5,
  description: "Lorem ipmsum",
};

test("renders correctly", () => {
  testSnapshot(<BeerListItem {...props} />);
});
