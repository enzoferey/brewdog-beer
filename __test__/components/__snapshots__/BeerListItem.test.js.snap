// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "background": "pink",
    }
  }
>
  <div
    className={undefined}
  >
    <img
      alt="Beer bottle image"
      src="src-path"
    />
  </div>
  <div
    className={undefined}
  >
    <h4
      className={undefined}
    >
      Beer name
    </h4>
    <span
      className={undefined}
    >
      Alc. 5% By Vol.
    </span>
    <p
      className={undefined}
    >
      Lorem ipmsum
    </p>
  </div>
</div>
`;
